chicago/events.py

```diff
@@ -29,40 +31,86 @@ def eventPages(self, event_url, search_type='all') :
                 when = when.replace(hour=event_time.hour)
 
+                status_string = location_list[-1].split('Chicago, Illinois')
+                if len(status_string) > 1 and status_string[1] :
+                    status_text = status_string[1].lower()
+                    if any(phrase in status_text 
+                           for phrase in ('rescheduled to',
+                                          'postponed to',
+                                          'reconvened to',
+                                          'recessed',
+                                          'cancelled',
+                                          'new date and time',
+                                          'rescheduled indefinitely',
+                                          'rescheduled for')) :
+                        status = 'cancelled'
+                    elif status_text in ('rescheduled') :
+                        status = 'cancelled'
+                    else :
+                        print(status_text)
+                elif datetime.datetime.utcnow().replace(tzinfo = pytz.utc) > when :
+                    status = 'confirmed'
+                else :
+                    status = 'passed'
+                            
+
                 e = Event(name=events["Name"]["label"],
-                          when=when,
+                          start_time=when,
+                          timezone='US/Central',
                           location=location,
                           status=status)
-                e.add_source(EVENTSPAGE)
-                if events['Video'] != u'Not\xa0available' :
-                    print(events['Video'])
+                e.add_source(detail_url)
+                if events['Video'] != 'Not\xa0available' : 
+                    e.add_media_link(note='Recording',
+                                     url = events['Video']['url'],
+                                     type="recording",
+                                     media_type = '???')
+
+                addDocs(e, events, 'Agenda')
+                addDocs(e, events, 'Notice')
+                addDocs(e, events, 'Transcript')
+                addDocs(e, events, 'Summary')
+
+                if events["Name"]["label"] != "City Council" :
+                    for item, _, _ in agenda :
+                        agenda_item = e.add_agenda_item(item["Title"])
+                        agenda_item.add_bill(item["Record #"]['label'])
+
+                
+                e.add_participant(name=events["Name"]["label"],
+                                  type="organization")
 
                 yield e
 
-
+def addDocs(e, events, doc_type) :
+    try :
+        if events[doc_type] != 'Not\xa0available' : 
+            e.add_document(note= events[doc_type]['label'],
+                           url = events[doc_type]['url'],
+                           media_type="application/pdf")
+    except ValueError :
```

Comment:
Hurm.

We have an `on_duplicate="ignore"` with Billy, we could implement something similar with Pupa, I guess. That or make the checking explicit or something